swarm/swarm.go

```diff
@@ -133,18 +138,18 @@ func NewSwarm(ctx *node.ServiceContext, backend chequebook.Backend, ensClient *e
 	self.dpa = storage.NewDPA(dpaChunkStore, self.config.ChunkerParams)
 	log.Debug(fmt.Sprintf("-> Content Store API"))
 
-	// set up high level api
-	transactOpts := bind.NewKeyedTransactor(self.privateKey)
-
-	if ensClient == nil {
-		log.Warn("No ENS, please specify non-empty --ens-api to use domain name resolution")
-	} else {
-		self.dns, err = ens.NewENS(transactOpts, config.EnsRoot, ensClient)
-		if err != nil {
-			return nil, err
+	if len(config.EnsAPIs) > 0 {
+		opts := []api.MultiResolverOption{}
+		for _, c := range config.EnsAPIs {
+			tld, endpoint, addr := parseEnsAPIAddress(c)
+			r, err := newEnsClient(endpoint, addr, config)
```

Comment:
I wonder if we should test for invalid formats before we actually try to connect to ENS? What do you think?

Reply:
Function parseEnsAPIAddress does not return error as the format is very simple, permissive in the endpoint part and assumes any value for tld and contract address. I would put format validation much sooner during execution, just after the config load in cmd/swarm.buildConfig. Would that be ok?

Reply:
Yeah I think that would be a good idea.